src/Extensions/SwiftypeSiteTreeCrawlerExtension.php

```diff
@@ -171,4 +200,60 @@ protected function getLogger(): LoggerInterface
     {
         return Injector::inst()->get(LoggerInterface::class);
     }
+
+
+    public function setUrlsToCrawl(array $urls) {
+        $this->urlsToCrawl = $urls;
+    }
+
+    public function getUrlsToCrawl(): array
+    {
+        return $this->urlsToCrawl;
+    }
+
+    private function getOwnerKey(): string
+    {
+        $owner = $this->getOwner();
+        $key = $owner->ClassName . $owner->ID;
+        return $key;
+    }
+
+    /**
+     * collate Urls to crawl
+     *
+     * @return void
+     */
+    public function collateUrls(): void
+    {
+        $urls = $this->getUrlsToCrawl();
+        $this->withVersionContext(function() use ($urls) {
+            /** @var SiteTree $owner */
+            $owner = $this->getOwner();
+            $key = $this->getOwnerKey();
+            $urls[$key] = [
+                $owner->AbsoluteLink(),
+            ];
```

Comment:
I don't think this link should be kept within an `array`?

You look through the keyed items in the `on..` methods, and will be passing through an `array` with one item each time to `forceSwiftypeIndex($url)`, and that's expecting a `string`?

```
foreach ($urls[$key] as $url)  {
    $this->forceSwiftypeIndex($url);
}
```

Reply:
I guess the idea with this is that you might want to update this in the future and add multiple urls - this just generalises the case so moving to multiple will be straight forward?

Reply:
Ok, cool.

Since we don't need this in a rush, I'm going to wait until we've got unit tests in here so that I can cover this there. It's a bit difficult to test otherwise :/